roles/ocp_netobserv/tasks/setup.yml

```diff
@@ -0,0 +1,215 @@
+---
+- name: Create the Namespace
+  community.kubernetes.k8s:
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: "netobserv"
+        labels:
+          security.openshift.io/scc.podSecurityLabelSync: "false"
+          pod-security.kubernetes.io/enforce: "baseline"
+          pod-security.kubernetes.io/enforce-version: "latest"
+
+- name: Setting Object Storage authentication
+  community.kubernetes.k8s:
+    definition:
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: netobserv-loki
+        namespace: "netobserv"
+      stringData:
+        access_key_id: "{{ netobserv_access_key_id | default(ocp_netobserv_access_key_id) }}"
+        access_key_secret: "{{ netobserv_access_key_secret | default(ocp_netobserv_access_key_secret) }}"
+        bucketnames: "{{ netobserv_bucket | default(ocp_netobserv_bucket) }}"
+        endpoint: "{{ netobserv_endpoint | default(ocp_netobserv_endpoint) }}"
+        region: "{{ netobserv_region | default(ocp_netobserv_region) }}"
+
+- name: Create the NetObservability Loki Stack
+  community.kubernetes.k8s:
+    definition:
+      apiVersion: loki.grafana.com/v1
+      kind: LokiStack
+      metadata:
+        name: netobserv-loki
+        namespace: "netobserv"
+      spec:
+        size: "{{ netobserv_loki_size | default(ocp_netobserv_loki_size) }}"
+        storage:
+          schemas:
+            - version: v12
+              effectiveDate: "{{ ansible_date_time.date }}"
+          secret:
+            name: netobserv-loki
+            type: s3
+        storageClassName: "{{ netobserv_storage_class | default(ocp_netobserv_storage_class) }}"
+        tenants:
+          mode: "openshift-network"
+
+- name: Configuring RBACs
+  community.kubernetes.k8s:
+    definition: |
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: ClusterRole
+      metadata:
+        name: netobserv-reader
+      rules:
+      - apiGroups:
+        - 'loki.grafana.com'
+        resources:
+        - network
+        resourceNames:
+        - logs
+        verbs:
+        - 'get'
+      ---
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: ClusterRole
+      metadata:
+        name: netobserv-writer
+      rules:
+      - apiGroups:
+        - 'loki.grafana.com'
+        resources:
+        - network
+        resourceNames:
+        - logs
+        verbs:
+        - 'create'
+      ---
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: ClusterRoleBinding
+      metadata:
+        name: netobserv-writer-flp
+      roleRef:
+        apiGroup: rbac.authorization.k8s.io
+        kind: ClusterRole
+        name: netobserv-writer
+      subjects:
+      - kind: ServiceAccount
+        name: flowlogs-pipeline
+        namespace: "netobserv"
+      - kind: ServiceAccount
+        name: flowlogs-pipeline-transformer
+        namespace: "netobserv"
+
+- name: Create the Flowcollector instance
```

Comment:
For reading purposes, isn't it better to move this kind of code to a template file and then just deploy the k8s resource based on the template in the Ansible task?